R/combine_bam.R

```diff
@@ -102,27 +102,29 @@
 bamtolist <- function(bamfolder, annotation, transcript_align = TRUE, 
                       name_samples = NULL, indel_threshold = 5, 
                       refseq_sep = NULL, granges = FALSE) {
-  names <- list.files(path = bamfolder, pattern = ".bam$")
+
+  name_bams <- list.files(path = bamfolder, pattern = ".bam$")
+
   if (length(name_samples) == 0) {
-    name_samples <- unlist(strsplit(names, ".bam"))
-    names(name_samples) <- unlist(strsplit(names, ".bam"))
+    name_samples <- unlist(strsplit(name_bams, ".bam"))
+    names(name_samples) <- unlist(strsplit(name_bams, ".bam"))
   } else {
-    if (length(name_samples) > length(names)) {
-      cat("\n")
-      stop("length of name_samples greater than number of files\n\n")
-    }
-    if (length(name_samples) < length(names)) {
-      cat("\n")
-      stop("length of name_samples smaller than number of files\n\n")
+    if(is.null(names(name_samples)) | any(names(name_samples) == "") ){
```

Comment:
I'm not sure the expression after the pipe, i.e.
`any(names(name_samples) == "")`
can ever return TRUE, unless one name of name_sample is specifically set to "" (unlikely).

Reply:
I agree that it is not very likely but it could happen when you append a non-named vector to a named vector.
For example, I define a sample name - file name vector, 
```
mysamples <- c(bamname1 = "sample1",
               bamname2 = "sample2")
```
and after a while I may forget that it has to be a named vector and want to add "bamname3" just temporarily by doing : 
```
bamtolist(..., name_samples = c(mysamples, "bamname3")  )
```
Then it looks for ".bam" instead of "bamname3.bam". In this case returning an error message that `name_samples` has to be a named vector would be more helpful, rather than returning "File not found: .bam".